src/test/scala/cromwell/CromwellTestkitSpec.scala

```diff
@@ -134,25 +134,38 @@ with DefaultTimeout with ImplicitSender with WordSpecLike with Matchers with Bef
     TestActorRef(new WorkflowManagerActor(dataAccess, new LocalBackend))
   }
 
-  def runWdl(fsm: TestFSMRef[WorkflowState, WorkflowActor.WorkflowFailure, WorkflowActor],
+  def runWdl(sampleWdl: SampleWdl,
              eventFilter: EventFilter,
-             expectedOutputs: Map[FullyQualifiedName, WdlValue] = Map.empty,
-             terminalState: WorkflowState = WorkflowSucceeded): Unit = {
-    assert(fsm.stateName == WorkflowSubmitted)
+             runtime: String = "",
+             terminalState: WorkflowState = WorkflowSucceeded): (TestActorRef[WorkflowManagerActor], WorkflowId) = {
+    val wma = buildWorkflowManagerActor(sampleWdl, runtime)
+    val submitMessage = WorkflowManagerActor.SubmitWorkflow(sampleWdl.wdlSource(runtime), sampleWdl.wdlJson, sampleWdl.rawInputs)
+    var workflowId: WorkflowId = null
     eventFilter.intercept {
-      fsm ! WorkflowActor.Start
       within(timeoutDuration) {
-        awaitCond(fsm.stateName == WorkflowRunning)
-        awaitCond(fsm.stateName == terminalState)
-
-        if (terminalState == WorkflowSucceeded)
-          fsm.stateData should be(WorkflowActor.NoFailureMessage)
-
-        if (terminalState == WorkflowFailed)
-          fsm.stateData.isInstanceOf[WorkflowActor.FailureMessage] shouldEqual true
+        workflowId = Await.result(wma.ask(submitMessage).mapTo[WorkflowId], timeoutDuration)
+        def workflowStatus = Await.result(wma.ask(WorkflowManagerActor.WorkflowStatus(workflowId)).mapTo[Option[WorkflowState]], timeoutDuration)
+        awaitCond(workflowStatus.contains(terminalState))
```

Comment:
I would prefer this be split into two statements, 1) waiting for _any_ terminal state, then 2) validating that the terminal state is the specific `terminalState` that's expected. Even better would be asserting that the correct error/non-error message is returned, as scalatest will then print that diff to the console.

Otherwise when the wrong terminal state is reached, the test throws a timeout, masking information.

Same goes for `runWdlAndAssertOutputs` below.